File: code/python/guess.py
import random

MAX_NUMBER = 100
MIN_NUMBER = 1

randomNumber = random.randint(MIN_NUMBER, MAX_NUMBER)
isPlaying = True

print("Guess a number between 1 and 100")

while isPlaying:
    guess = int(input("Your guess: "))
    if guess == randomNumber:
        print("You guessed it!")
        isPlaying = False
    elif guess > randomNumber:
        print("Too high")
    else:
        print("Too low")

File: code/javascript/guess.mjs
import * as readlinePromises from 'node:readline/promises';
const rl = readlinePromises.createInterface({input: process.stdin,output: process.stdout});

const MAX_NUMBER = 100;
const MIN_NUMBER = 1;
const randomNumber = Math.floor(Math.random() * (MAX_NUMBER - MIN_NUMBER + 1)) + MIN_NUMBER;

let isPlaying = true;

console.log('Guess a number between 1 and 100');

while(isPlaying){
    const answer = await rl.question('Your guess: ');
    const number = parseInt(answer);

    if (number === randomNumber) {
        console.log('You guessed it!');
        isPlaying = false;
    }
    else if (number < randomNumber) {
        console.log('Too low!');
    }   else {  
        console.log('Too high!');
    }
} 


process.exit(0);